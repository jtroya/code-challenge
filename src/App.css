body {
  margin: 0;
  padding: 0; }

.container div {
  font-family: monospace;
  font-size: 14px;
  text-align: center; }

.header {
  background-color: #ff7d7d; }

.footer {
  background-color: #a6eaa6; }

.App {
  display: flex;
  min-height: 100vh;
  flex-direction: column; }

.body {
  display: flex;
  flex: 1; }

.container {
  display: flex;
  flex-wrap: wrap; }

.header {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 60px; }

.footer {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 60px; }

.box {
  display: block;
  border: 1px solid #c380c3;
  margin: 10px;
  padding: 9px 9px; }

.box-excerpt {
  font-size: 12px;
  text-align: left; }

/* Mobile Styles */
@media only screen and (max-width: 400px) {
  .container {
    justify-content: center; }
  .box {
    flex-basis: 100%; } }

/* Tablet Styles */
@media only screen and (min-width: 401px) and (max-width: 960px) {
  .container {
    justify-content: flex-start; }
  .box {
    flex-basis: 25%; } }

/* Desktop Styles */
@media only screen and (min-width: 961px) {
  .container {
    justify-content: flex-start; }
  .box {
    flex-basis: 20%; } }
