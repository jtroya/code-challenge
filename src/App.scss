// Shared Variables
$headerHeight: 60px;
$footerHeight: 60px;
$boxHeight: 100px;
$itemsPerRow: 5;

body {
  margin: 0;
  padding: 0;
}
  
.container div {
  font-family: monospace;
  font-size: 14px;
  text-align: center;
}
  
.header {
  background-color: #ff7d7d;
}

.footer {
  background-color: #a6eaa6;
}
  
.App {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.body {
  display: flex;
  flex: 1;
}

.container {
  display: flex;
  flex-wrap: wrap;
}
  
.header {
  // flex: 0 0 auto;
  // flex-basis: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  height: $headerHeight;
}
  
.footer {
  // flex-basis: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  height: $footerHeight;
}

.box {
  display: block;
  // align-items: center;
  // justify-content: center;
  border: 1px solid #c380c3;
  margin: 10px;
  padding: 9px 9px;
}

.box-excerpt {
  font-size: 12px;
  text-align: left;
}

/* Mobile Styles */
@media only screen and (max-width: 400px) {
  .container {
      justify-content: center;
  }

  .box {
    flex-basis: 100%;
  }
}

/* Tablet Styles */
@media only screen and (min-width: 401px) and (max-width: 960px) {
  .container {
    justify-content: flex-start;
  }

  .box {
    flex-basis: 25%;
  }
}

/* Desktop Styles */
@media only screen and (min-width: 961px) {
  .container {
    justify-content: flex-start;
  }

  .box {
    flex-basis: 100% / $itemsPerRow;
  }
}