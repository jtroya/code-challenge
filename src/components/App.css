body {
  margin: 0;
  padding: 0;
  font-family: "avenir next", "avenir"; }

.container div {
  font-size: 14px;
  text-align: center;
  padding: 10px; }

.header {
  background-color: #2196F3;
  color: #fff; }

.footer {
  background-color: #2196F3;
  color: #fff; }

.App {
  display: flex;
  min-height: 100vh;
  flex-direction: column; }

.body {
  background-color: #E3F2FD;
  display: flex;
  flex: 1; }

.container {
  display: flex;
  flex-wrap: wrap; }

.container-article {
  display: block; }

.header {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 60px; }

.footer {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 60px; }

.box {
  background-color: #BBDEFB;
  display: block;
  border: 1px solid #BBDEFB;
  margin: 10px;
  padding: 9px 9px; }

.box-excerpt {
  font-size: 12px;
  text-align: left;
  padding: 0px 12px; }

.box-info {
  margin: 10px;
  padding: 10px 10px; }

.loading {
  text-align: center;
  flex: initial; }

.link-article {
  color: #333; }

/* Mobile Styles */
@media only screen and (max-width: 400px) {
  .container {
    justify-content: center; }
  .box {
    flex-basis: 100%; } }

/* Tablet Styles */
@media only screen and (min-width: 401px) and (max-width: 960px) {
  .container {
    justify-content: flex-start; }
  .box {
    flex-basis: 25%; } }

/* Desktop Styles */
@media only screen and (min-width: 961px) {
  .container {
    justify-content: flex-start; }
  .box {
    flex-basis: 20%; } }
