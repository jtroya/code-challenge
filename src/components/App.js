import React, { Component } from 'react';
import { BrowserRouter, Route } from 'react-router-dom';

import ArticleListContainer from '../containers/ArticleListContainer';
import ArticleDetailContainer from '../containers/ArticleDetailContainer';

class App extends Component {
  // Renders
  render() {
    return (
      <BrowserRouter>
        <div className="App">
          <div className="header">
            <h1>Header</h1>
          </div>
          <div className="body">
            <Route exact path="/" component={ArticleListContainer} />
            <Route
              exact
              path="/article/:id"
              render={({ match }) => (
                <ArticleDetailContainer id={match.params.id} />
              )}
            />
          </div>
          <div className="footer">Footer</div>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
