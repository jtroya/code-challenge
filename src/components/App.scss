// Shared Variables
$headerHeight: 60px;
$footerHeight: 60px;
$boxHeight: 100px;
$itemsPerRow: 5;
$fonts: 'avenir next', 'avenir';
$bkgColor: #2196F3;
$bodyColor: #E3F2FD;
$boxBkgColor: #BBDEFB;
$textColor: #fff;
$linkColor: #333;

// $bkgColor: #242819;
// $bodyColor: #d9d3e7;
// $boxBkgColor: #bfbac9;
// $textColor: #d9d3e7;

body {
  margin: 0;
  padding: 0;
  font-family: $fonts;
}
  
.container div {
  font-size: 14px;
  text-align: center;
  padding: 10px;
}
  
.header {
  background-color: $bkgColor;
  color: $textColor;
}

.footer {
  background-color: $bkgColor;
  color: $textColor;
}
  
.App {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.body {
  background-color: $bodyColor;
  display: flex;
  flex: 1;
}

.container {
  display: flex;
  flex-wrap: wrap;
}

.container-article {
  display: block;
}
  
.header {
  display: flex;
  align-items: center;
  justify-content: center;
  height: $headerHeight;
}
  
.footer {
  display: flex;
  align-items: center;
  justify-content: center;
  height: $footerHeight;
}

.box {
  background-color: $boxBkgColor;
  display: block;
  border: 1px solid $boxBkgColor;
  margin: 10px;
  padding: 9px 9px;
}

.box-excerpt {
  font-size: 12px;
  text-align: left;
  padding: 0px 12px;
}

.box-info {
  margin: 10px;
  padding: 10px 10px;
}

.loading {
  text-align: center;
  flex: initial;
}

.link-article {
  color: $linkColor;
}

/* Mobile Styles */
@media only screen and (max-width: 400px) {
  .container {
      justify-content: center;
  }

  .box {
    flex-basis: 100%;
  }
}

/* Tablet Styles */
@media only screen and (min-width: 401px) and (max-width: 960px) {
  .container {
    justify-content: flex-start;
  }

  .box {
    flex-basis: 25%;
  }
}

/* Desktop Styles */
@media only screen and (min-width: 961px) {
  .container {
    justify-content: flex-start;
  }

  .box {
    flex-basis: 100% / $itemsPerRow;
  }
}