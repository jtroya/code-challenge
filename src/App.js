import React, { Component } from 'react';

import request from './request';
import { ARTICLES_QUERY } from './queries';
import ArticleList from './ArticleList';
import './App.css';

class App extends Component {
  // definition
  constructor(props) {
    super(props);
    this.state = {
      articles: [],
    };
  }

  // lifecycle
  componentWillMount() {
    request(ARTICLES_QUERY).then(response => {
      this.setState({ articles: response.data.articles });
    });
  }

  // Renders
  render() {
    // console.info('data', JSON.stringify(this.state.articles, null, 2));
    const { articles } = this.state;
    return (
      <div className="App">
        <div className="header">
          <h2>Header</h2>
        </div>
        <div className="body">
          <ArticleList items={articles} />
        </div>
        <div className="footer">Footer</div>
      </div>
    );
  }
}

export default App;
