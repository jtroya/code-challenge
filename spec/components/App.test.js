import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from '../../src/components/App';

describe('App', () => {
  let wrapper;

  it('should render de App', () => {
    wrapper = shallow(<App />);
    expect(wrapper.find('.App')).toBeTruthy();
    expect(wrapper.find('.header')).toBeTruthy();
    expect(wrapper.find('ArticleDetailContainer')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
