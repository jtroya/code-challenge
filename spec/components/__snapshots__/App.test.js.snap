exports[`App matches the snapshot 1`] = `
<BrowserRouter>
  <div
    className="App">
    <div
      className="header">
      <h1>
        Header
      </h1>
    </div>
    <div
      className="body">
      <Route
        component={[Function]}
        exact={true}
        path="/" />
      <Route
        exact={true}
        path="/article/:id"
        render={[Function]} />
    </div>
    <div
      className="footer">
      Footer
    </div>
  </div>
</BrowserRouter>
`;
